asm/additions/rust-additions/src/traps.rs

```diff
@@ -156,15 +158,18 @@ pub fn update_traps() {
                     z: 1.0,
                 } as *mut math::Vec3f;
 
-                actor::spawn_actor(
-                    actor::ACTORID::NPC_RVL,
-                    (*ROOM_MGR).roomid.into(),
-                    0xFFFFFFFF,
-                    actor_pos,
-                    actor_rot,
-                    actor_scale,
-                    0xFFFFFFFF,
-                );
+                // Only spawn Groose if not in a Silent Realm because it lags for 2-3 seconds
+                if &CURRENT_STAGE_NAME[..1] != b"S" {
```

Comment:
```suggestion
                if &CURRENT_STAGE_NAME[..1] != b"S" || &CURRENT_STAGE_NAME[..7] == b"D003_8\0" {
```
Should add a check for the Sky Keep silent realm map also.